app/BUILD.bazel

```diff
@@ -387,6 +387,9 @@ VIEWS = [
 BINDING_ADAPTERS_WITH_RESOURCE_IMPORTS = [
     "src/main/java/org/oppia/android/app/databinding/DrawableBindingAdapters.java",
     "src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java",
+    "src/main/java/org/oppia/android/app/databinding/MarginBindingAdapters.java",
+    "src/main/java/org/oppia/android/app/databinding/StateAssemblerMarginBindingAdapters.java",
+    "src/main/java/org/oppia/android/app/databinding/StateAssemblerPaddingBindingAdapters.java",
```

Comment:
I think this should go into the `BINDING_ADAPTERS` only, but that's also giving the error that ViewCompat is coming from external dependency. 

Also, we need to add these new Test files to `kt_android_library` naming `test_deps`. 
Till now we had tried these things separately, lets fix them all and check if the build passes or not. 

Also, do we need any kind of dependency inside `TEST_DEPS`, is there anything currently which provides ViewCompat, 
I think the dependency should be `androidx.core:core-ktx:1.0.2`, which is there for android library `databinding_resources`, but not for the tests.

Reply:
I have updated the changes and pushed the code. Let me see if it is resolved.

Reply:
@veena14cs you need to include the viewcompact in the [deps](https://github.com/oppia/oppia-android/blob/develop/app/BUILD.bazel#L493) for the BINDING_ADAPTERS
I think the dependency is
```
"//third_party:androidx_core_core"
```

@anandwana001 I didn't get the ktx part here, we are using java right, does java support that?

Reply:
Ah it is already added

Reply:
@FareesHussain yes right. Did you get what is missing.?

Reply:
I'm unable to figure it out.
@fsharpasharp, @BenHenning any idea when do we get the following error
```
app/src/main/java/org/oppia/android/app/databinding/MarginBindingAdapters.java:97: error: [strict] Using type androidx.core.view.ViewCompat from an indirect dependency (TOOL_INFO: "bazel-out/android-armeabi-v7a-fastbuild/bin/external/maven/_aar/androidx_core_core/classes_and_libs_merged.jar").
    return ViewCompat.getLayoutDirection(view) == ViewCompat.LAYOUT_DIRECTION_RTL;
``` 

I don't understand when indirect dependency mean here.